### ISSUE TITLE
Can filter tests randomly fail

### ISSUE THREAD

**Issue by bufke** [burke-software/django-report-builder#187]
@AbhiAgarwal this is really strange. If I run the unit tests about 10 times - I'll produce a failure.

```
FAIL: test_report_builder_can_filter (report_builder.tests.ReportBuilderTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/report_builder/tests.py", line 206, in test_report_builder_can_filter
    self.assertEqual(field['can_filter'], False)
AssertionError: True != False
```

It's fairly rare. It could be a race condition. I was able to capture which field it fails on.

```
{'name': 'i_want_char_field', 'field': 'i_want_char_field', 'field_verbose': 'i_want_char_field', 'field_choices': None, 'path_verbose': '', 'help_text': 'Adding this property will significantly increase the time it takes to run a report.', 'is_default': True, 'path': '', 'can_filter': True, 'field_type': 'Property'}
```

This only happens because I improved the unit test - but I think it's uncovering a potentially real issue. Given how transient it is - it's hard to debug. Any ideas? I'd like to fix this before making a new pypi release.

**Comment by AbhiAgarwal** [comment]
@bufke I will check out this the first thing tomorrow morning. I'll try do a trace, and see what is happening.

**Comment by grantjenks** [comment]
I ran `python manage.py test` for 2 hours this morning (1,079 times) and it passed every time. I'm afraid I'm no help here.

**Comment by AbhiAgarwal** [comment]
I ran this yesterday as well, and I couldn't find any other errors for a significant number of test-runs. The only errors that I managed to find were fixed in my pull-request (#188).
